hcipy/plotting/animation.py

```diff
@@ -1,43 +1,83 @@
-import matplotlib.pyplot as plt
+import glob
+import os
+from subprocess import Popen, PIPE
+
 import matplotlib
-from subprocess import Popen, PIPE, call
 import imageio
-import os
+from PIL import Image
+
 
 class GifWriter(object):
 	def __init__(self, filename, framerate=15):
 		self.closed = False
 		self.filename = filename
 		self.framerate = framerate
 
-		if not os.path.exists(filename + '_frames/'):
-			os.mkdir(filename + '_frames/')
-		self.frame_number = 0
-	
+		self.path_to_frames = self.filename + "_frames"
+		if not os.path.exists(self.path_to_frames):
+			os.mkdir(self.path_to_frames)
+		self.n_frames = 0
+
+	def __del__(self):
+		try:
+			self.close()
+		except Exception:
+			pass
+
 	def add_frame(self, fig=None, arr=None, cmap=None):
 		if self.closed:
 			raise RuntimeError('Attempted to add a frame to a closed GifWriter.')
-		
-		dest = self.filename + '_frames/%05d.png' % self.frame_number
-		self.frame_number += 1
+
+		dest = os.path.join(self.path_to_frames, '%05d.png' % self.n_frames)
 
 		if arr is None:
 			if fig is None:
-				fig = plt.gcf()
+				fig = matplotlib.pyplot.plt.gcf()
```

Comment:
This won't work. Remove the plt.

Reply:
Also, the same change needs to be made in the `add_frame()` of `FFMpegWriter` below to not break that function.